bz18883: remove progress bar when importing metadata.

bendk opened the pull request:
Refactored things slightly now that we only have 2 modes of display, but
overall I tried not to change too much.

Commit:
bz18883: remove progress bar when importing metadata.

Refactored things slightly now that we only have 2 modes of display, but
overall I tried not to change too much.

paulswartz commented:
Kind of related to this; there's a pyflakes error in tv/lib/frontends/widgets/itemlistcontroller.py:138

``` python
    def update_metadata_progress(self, finished, finished_local, eta, total):
        meter = self.widget.get_progress_meter()
        if meter:
            meter.update(self.mediatype, finished, finished_local, eta, total)
            self.postponed = None
        else: # got progress before widget created
            self.postponed = (remaining, eta, total)
```

`self.postponed` should be set to the same arguments as the update call, possibly like this?

``` python
    def update_metadata_progress(self, finished, finished_local, eta, total):
        meter = self.widget.get_progress_meter()
        args = (self.mediatype, finished, finished_local, eta, total)
        if meter:
            meter.update(*args)
            self.postponed = None
        else: # got progress before widget created
            self.postponed = args
```

Commit:
Related fix: set postponed correctly.

The signature for update_metadata_progress() changed, but we didn't update
postponed.

bendk commented:
Good catch on the that one.  I just added a changeset to address it.

geoffl commented:
looks good
